etna/transforms/math/sklearn.py

```diff
@@ -227,8 +227,8 @@ def _inverse_transform(self, df: pd.DataFrame) -> pd.DataFrame:
 
     def _preprocess_macro(self, df: pd.DataFrame) -> np.ndarray:
         segments = sorted(set(df.columns.get_level_values("segment")))
-        x = df.loc[:, pd.IndexSlice[:, self.in_column]]
-        x = pd.concat([x[segment] for segment in segments]).values
+        x = df.loc[:, pd.IndexSlice[segments, self.in_column]]
+        x = TSDataset.to_flatten(x)[self.in_column].values
```

Comment:
It might be faster to slice inside `to_flatten` as it transform only part of the dataframe inside
```
x = TSDataset.to_flatten(x, features=self.in_column)[self.in_column].values
```